### PR TITLE
Finished

### PR DESCRIPTION
**Pull request by Jcmcn**

### CODE REVIEW

File: Problem2/src/problem2/Problem2.java [hunk 1]
```diff
@@ -15,7 +15,28 @@
      * @param args the command line arguments
      */
     public static void main(String[] args) {
-        // TODO code application logic here
-    }
+        
+        }
     
+    public boolean division(long n){
```
Comment (cgallinaro) [inline Problem2/src/problem2/Problem2.java:21]: Your algorithm is not recursive.

File: Problem3/src/problem3/Person.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
  * 
  * @author cgallinaro
  */
-public class Person {
+public class Person implements Comparable<Person>{
```
Comment (cgallinaro) [inline Problem3/src/problem3/Person.java:9]: No insertion sort implementation.

No tests.

File: Problem4/src/problem4/Problem4.java [hunk 3]
```diff
@@ -4,18 +4,77 @@
...
+     * @param array The unsorted array
+     * @return sortedArray The array after being sorted
+     * 
      */
-    public static void main(String[] args) {
-        // TODO code application logic here
+    public int[] numberSorting(int[] array){
+        
+        if (array.length == 0){
+            return null;
```
Comment (cgallinaro) [inline Problem4/src/problem4/Problem4.java:40]: Why return null if the array length is 0?  Should just return the empty array.

File: Problem4/src/problem4/Problem4.java [hunk 4]
```diff
@@ -4,18 +4,77 @@
...
+        }
+        
+        int higherPlace = array.length - 1; // Variable used to set the highest available place in the array
+        int lookingForHigher = 0; //Variable used to loop through all components in the array
+        int lowestNumber = array[0]; //Variable used to determine the lowest value in the array
+        
+        for (int i = 0; i < array.length; i++) {
```
Comment (cgallinaro) [inline Problem4/src/problem4/Problem4.java:47]: Not sure this for loop is required

File: Problem4/src/problem4/Problem4.java [hunk 5]
```diff
@@ -4,18 +4,77 @@
...
+            int tempValue = array[higherPlace];
+            array[higherPlace] = topPick;
+            array[originalPosition] = tempValue;
+            lookingForHigher = 0;
+
+            topPick = lowestNumber;
+            higherPlace--;
+        }
+
+        int[] sortedArray = array.clone();
```
Comment (cgallinaro) [inline Problem4/src/problem4/Problem4.java:76]: No need to clone the array that you are returning.  In fact, since you are modifying the passed-in array, your method should not return anything.

File: Problem4/test/problem4/Problem4Test.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        Assert.assertArrayEquals(test4Sorted, test.numberSorting(test4NotSorted));
+        
+    }
+    
+    
+    
+    
+    
+    @Test
+    public void mixtestnumberSorting(){
```
Comment (cgallinaro) [inline Problem4/test/problem4/Problem4Test.java:79]: Good edge case coverage.

File: Problem5/test/problem5/Problem5Test.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.Test;
+
+/**
+ *
+ * @author Joca
+ */
+public class Problem5Test {
+    
+    @Test
+    public void testbinarySearchRecursive(){
```
Comment (cgallinaro) [inline Problem5/test/problem5/Problem5Test.java:18]: Missing test coverage for a word not in the list.  Should also have additional tests that don't use the given word list.  It would allow you cover edge cases like:
- empty array
- single word array
- word list where there are no letter starting with a particular character (i.e. no words that start with 'c')

**Comment by cgallinaro** [comment]
Problem 1: 5/5
Problem 2: 4/5
Problem 3: 1/5
Problem 4: 9/10
Problem 5: 9/10

Total: 28/35
